### PR TITLE
feat: Add schedule scaling possibility

### PR DESCRIPTION
**Pull request by mfroembgen**
## Description
Additionally, it would be nice to use schedule scaling for ddb tables. (https://docs.aws.amazon.com/autoscaling/application/userguide/get-started-exercise.html)
If you want to save cost with *PROVISIONED* but have for example an import job that needs a lot of capacity it's the only option to not run into scaling issues. 

## Motivation and Context
If you want to save cost with *PROVISIONED* but have for example an import job that needs a lot of capacity it's the only option to not run into scaling issues. 

## Breaking Changes
No, as it is only an additional option. 

## Draft
This is just a draft of how It could be implemented. I'm open to superior solutions as I know its kinda bad practice. 

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by mfroembgen** [comment]
@antonbabenko Do you maybe have time to share your opinions on that, please?

File: schedulescaling.tf [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  schedule = each.value["schedule"]
+
+  scalable_target_action {
+    max_capacity = each.value["max_capacity"]
+    min_capacity = each.value["min_capacity"]
+  }
+}
+
+module "index_read_schedule" {
+  for_each = var.create_table && length(var.schedule_scaling_indexes_read) > 0 ? var.autoscaling_indexes : {}
```
Comment (antonbabenko) [inline schedulescaling.tf:19]: Looks pretty good, but please remove `module/module` and implement the same logic in the existing `autoscaling.tf`

File: schedulescaling.tf [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  resource_id        = aws_appautoscaling_target.index_read[each.key].resource_id
+  scalable_dimension = aws_appautoscaling_target.index_read[each.key].scalable_dimension
+  service_namespace  = aws_appautoscaling_target.index_read[each.key].service_namespace
+
+  schedule_scaling_indexes = var.schedule_scaling_indexes_read[each.key]
+}
+
+resource "aws_appautoscaling_scheduled_action" "table_write_schedule" {
+  for_each = var.create_table && var.autoscaling_enabled && length(var.schedule_scaling_write) > 0 ? { for k, v in var.schedule_scaling_write : k => v } : {}
```
Comment (antonbabenko) [inline schedulescaling.tf:32]:
```suggestion
  for_each = { for k, v in var.schedule_scaling_write : k => v if var.create_table && var.autoscaling_enabled && length(var.schedule_scaling_write) > 0}
```

and the same in other `for_each`

File: variables.tf [hunk 3]
```diff
@@ -156,6 +156,46 @@ variable "autoscaling_indexes" {
   default     = {}
 }
 
+variable "schedule_scaling_read" {
```
Comment (antonbabenko) [inline variables.tf:159]: Please change the variable names, type and default to match existing variables like `autoscaling_indexes`

Comment (antonbabenko) [reply variables.tf:159]:
```suggestion
variable "autoscaling_scheduled_read" {
```

File: variables.tf [hunk 4]
```diff
@@ -156,6 +156,46 @@ variable "autoscaling_indexes" {
...
+  description = "A map of read schedule scaling settings. See example in examples/autoscaling"
```
Comment (antonbabenko) [inline variables.tf:160]:
```suggestion
  description = "A map of read scheduled scaling settings. See example in examples/autoscaling"
```

File: schedulescaling.tf (hunk 2 above)
Comment (antonbabenko) [reply schedulescaling.tf:32]: Please add support for available arguments of `aws_appautoscaling_scheduled_action` resource - `start_time`, `end_time`, `timezone`.

Ref - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appautoscaling_scheduled_action#start_time

**Review by antonbabenko — CHANGES_REQUESTED** [review]

File: schedulescaling.tf (hunk 2 above)
Comment (mfroembgen) [reply schedulescaling.tf]: Added as optional values

File: variables.tf (hunk 3 above)
Comment (mfroembgen) [reply variables.tf]: Adjusted

File: variables.tf (hunk 4 above)
Comment (mfroembgen) [reply variables.tf]: fixed

**Comment by mfroembgen** [comment]
@antonbabenko Can you maybe have another look at it, please? Please let me know what you think 🙂

**Comment by mfroembgen** [comment]
@antonbabenko Just a friendly reminder that I'm still looking for another review :)

**Pull request closed without merging**